src/utils.jl

```diff
@@ -34,10 +34,14 @@ function reverse_indices!(rev::AbstractArray, idx::AbstractArray)
     rev
 end
 
-function reverse_indices(idx::AbstractArray)
-    rev = Array{Vector{CartesianIndex}}(undef, maximum_dims(idx)...)
+reverse_indices(idx::AbstractArray) = _reverse_indices(idx)
```

Comment:
doesn't seem we need `_reverse_indices`, we can just move the implementation to `rerverse_indices` itself

Also, it would be good to add a docstring

Reply:
I put another API for `reverse_indices(idx::AnyCuArray)`, and in that API it calls `_reverse_indices(idx)`.

Reply:
I think it's fine you copy the implementation in NNlibCUDA since it is quite short, this way we rely less on internals and it also becomes quite clear that we are going through cpu operations and at some point we will need a proper kernel

Reply:
Done.